kafka/admin/client.py

```diff
@@ -484,28 +485,74 @@ def describe_configs(self, config_resources, include_synonyms=False):
-                    .format(self.config['api_version']))
-            request = DescribeConfigsRequest[version](
-                resources=[self._convert_describe_config_resource_request(config_resource) for config_resource in config_resources]
-            )
+                        .format(self.config['api_version']))
+
+            if len(broker_resources) > 0:
+                for broker_resource in broker_resources:
+                    try:
+                        futures.append(self._send_request_to_node(
+                            int(broker_resource[1]),
+                            DescribeConfigsRequest[version](resources=[broker_resource])
+                        ))
+                    except ValueError:
+                        raise ValueError("Broker resource names must be an integer or a string represented integer")
+
+            if len(topic_resources) > 0:
+                futures.append(self._send_request_to_node(
+                    self._client.least_loaded_node(),
+                    DescribeConfigsRequest[version](resources=topic_resources)
+                ))
+
         elif version == 1:
-            request = DescribeConfigsRequest[version](
-                resources=[self._convert_describe_config_resource_request(config_resource) for config_resource in config_resources],
-                include_synonyms=include_synonyms
-            )
+            if len(broker_resources) > 0:
+                for broker_resource in broker_resources:
+                    try:
+                        futures.append(self._send_request_to_node(
+                            int(broker_resource[1]),
+                            DescribeConfigsRequest[version](
+                                resources=[broker_resource],
+                                include_synonyms=include_synonyms)
+                        ))
+                    except ValueError:
+                        raise ValueError("Broker resource names must be an integer or a string represented integer")
+
+            if len(topic_resources) > 0:
+                futures.append(self._send_request_to_node(
+                    self._client.least_loaded_node(),
+                    DescribeConfigsRequest[version](resources=topic_resources, include_synonyms=include_synonyms)
+                ))
         else:
             raise NotImplementedError(
                 "Support for DescribeConfigs v{} has not yet been added to KafkaAdminClient."
-                .format(version))
-        future = self._send_request_to_node(self._client.least_loaded_node(), request)
+                    .format(version))
+
+        self._wait_for_futures(futures)
+
+        # Use one of the results as the general response and add all other resources to it
+        response = copy.copy(futures[0])
+        response.resources = []
+
+        for future in futures:
+            response.resources.extend(future.value.resources)
```

Comment:
This seems a bit strange to me. If we want to combine responses into a singular return value, why keep the response structure? I would prefer to only return response data classes when they are unmodified.

Reply:
I agree, this was a bit odd. The reason for combining the responses into one was to keep the same response format. As it is now (with the above fix of copying the value attribute instead of the future object) An alternative would be to return a list of responses, but that would break contract. 

I am open for suggestions here :-)

Reply:
> An alternative would be to return a list of responses, but that would break contract.

Breaking changes are fine in `KafkaAdminClient`--I explicitly put a big warning in the docs about that. Not what we want long term, but right now there are a number of methods that are only partially stubbed out (like this one) and so I decided the simplest course was just say "you can use this, just beware it may change" as the other alternative was either not releasing it until fully complete, or tying our hands into awkward/broken interfaces.

So do whatever seems best, w/o regard to backward compatibility right now.

Reply:
I simplified it to just return a list of the describe config responses. For future work it could be converted into a more custom type similar to the ACL